exporter/elasticsearchexporter/factory.go

```diff
@@ -120,8 +132,10 @@ func createLogsExporter(
 		exporter.pushLogsData,
 		exporterhelper.WithCapabilities(consumer.Capabilities{MutatesData: true}),
 		exporterhelper.WithStart(exporter.Start),
+		exporterhelper.WithBatcher(cf.BatcherConfig),
 		exporterhelper.WithShutdown(exporter.Shutdown),
 		exporterhelper.WithQueue(cf.QueueSettings),
+		exporterhelper.WithTimeout(getTimeoutConfig()),
```

Comment:
Can we make it similar to this? https://github.com/open-telemetry/opentelemetry-collector/blob/v0.104.0/exporter/otlphttpexporter/factory.go#L90-L91

I think the comment better describes what is happening, and we can simplify the code getting rid of the `getTimeoutConfig` function.

```suggestion
		// explicitly disable since we rely on the timeout defined in httpClientConfig.Timeout.
		exporterhelper.WithTimeout(exporterhelper.TimeoutSettings{Timeout: 0}),
```